app/src/main/java/com/bunbeauty/papakarlo/feature/auth/screen/login/LoginViewModel.kt

```diff
@@ -13,27 +13,40 @@ import com.bunbeauty.shared.Constants.PHONE_CODE
 import com.bunbeauty.shared.Constants.TOO_MANY_REQUESTS
 import com.bunbeauty.shared.domain.interactor.user.IUserInteractor
 import com.google.firebase.auth.PhoneAuthProvider
+import kotlinx.coroutines.CoroutineExceptionHandler
 import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 
 class LoginViewModel(
     private val textValidator: ITextValidator,
     private val userInteractor: IUserInteractor,
     private val firebaseAuthRepository: FirebaseAuthRepository,
-    savedStateHandle: SavedStateHandle
+    savedStateHandle: SavedStateHandle,
 ) : BaseViewModel() {
 
-    private val mutableIsLoading: MutableStateFlow<Boolean> = MutableStateFlow(false)
-    val isLoading: StateFlow<Boolean> = mutableIsLoading.asStateFlow()
+    private val mutableLoginState = MutableStateFlow(LoginState())
+    val loginState = mutableLoginState.asStateFlow()
+
+    private val exceptionHandler = CoroutineExceptionHandler { coroutineContext, throwable ->
+        mutableLoginState.update { oldState ->
+            oldState.copy(
+                state = LoginState.State.Error(throwable)
+            )
+        }
+    }
 
     private val successLoginDirection: SuccessLoginDirection by lazy {
         savedStateHandle["successLoginDirection"] ?: BACK_TO_PROFILE
     }
 
     fun setNotLoading() {
```

Comment:
Not entirely clear naming. Please rename it to `setSuccessState()` for example.

Reply:
+